src/StateSmith/Output/Sim/HtmlRenderer.cs

```diff
@@ -376,17 +375,18 @@ public static void Render(StringBuilder stringBuilder, string smName, string moc
         };
 
         // Wire up the buttons that dispatch events for the state machine.
-        for (const eventName in {{smName}}.EventId) {
+        diagramEventNamesArray.forEach(diagramEventName => {
             var button = document.createElement('button');
-            button.id = 'button_' + eventName;
-            button.innerText = eventName;
+            button.id = 'button_' + diagramEventName;
+            button.innerText = diagramEventName;
             button.addEventListener('click', () => {
                 clearHighlightedEdges();
-                sm.tracer.log(""Dispatched "" + eventName, true);
-                sm.dispatchEvent({{smName}}.EventId[eventName]); 
+                sm.tracer.log(""Dispatched "" + diagramEventName, true);
```

Comment:
nit: suggest sm.tracer?.log

Reply:
ah, i see your TODO comment below about removing conditional.
Agree we set the tracer in the simulator so it wouldn't hurt anything to get rid of the conditional.
My thought process is that generated code shouldn't assume the tracer is set since the "user" could do all kinds of things, like unset it. That being said, the "user" is us, so it doesn't make a difference in practice. But once we migrate the tracer code to the JS codegen, we'll need to use the conditional.

Reply:
I don't mind adding the conditional. It doesn't hurt anything right now and it may be handy in the future: https://github.com/StateSmith/StateSmith/pull/311/commits/0ce4d003cee1f4dc766c73972b4764616fb22997

When we do add the built in tracer, it likely will look like code that can be transpiled to C. Instead of seeing `this.tracer?.enterState(some args)`, it will look like `if (this.tracer != null) this.tracer.enterState(some args`

Reply:
tangent about build in tracer: Hmm... we might want to check for enterState being null too. 
```c#
if (this.tracer != null && this.tracer.enterState != null) {
    this.tracer.enterState(some args);
}
```
Or instruct user that all tracer function pointers must be non-null even if the user doesn't use them. I could see someone maybe not caring about exitState for example. They would have to point it to a non-null dummy function.